critiquebrainz/db/revision.py

```diff
@@ -148,23 +152,34 @@ def get_revision_number(review_id, revision_id):
     return rev_num
 
 
-def create(review_id, text):
+def create(review_id, text, rating):
     """Creates a new revision for the given review.
 
     Args:
         review_id (uuid): ID of the review.
-        text (str): Updated/New text of the review.
+        text (str): Updated/New text part of the review.
+        rating (int): Updated/New rating part of the review
     """
     with db.engine.connect() as connection:
         connection.execute(sqlalchemy.text("""
-            INSERT INTO revision(review_id, timestamp, text)
-                 VALUES (:review_id, :timestamp, :text)
+            INSERT INTO revision(review_id, timestamp, text, rating)
+                 VALUES (:review_id, :timestamp, :text, :rating)
         """), {
             "review_id": review_id,
             "timestamp": datetime.now(),
             "text": text,
+            "rating": rating,
         })
 
+    review = db_review.get_by_id(review_id)
+    rev_num = get_revision_number(review["id"], review["last_revision"]["id"])
+    if rev_num > 1:
+        revisions = get(review["id"], limit=2, offset=0)
+        if revisions[0]["rating"] != revisions[1]["rating"]:
+            db_avg_rating.update(review["entity_id"], review["entity_type"])
+    elif rating is not None:
+        db_avg_rating.update(review["entity_id"], review["entity_type"])
```

Comment:
I don't think I understand this part, can you clarify what it does? It also seems like that should go into a separate function.